File: angularweb/src/app/model/order/order-dto.ts
import {OrderItemDto} from "./order-item-dto";

export class OrderDto {
  id: number = 0;
  userId: number = 0;
  status: string = '';
  total: number = 0;
  orderDate: string = '';
  shippingAddress: string = '';
  orderItems: OrderItemDto[] = [];
}

File: src/main/java/webapplication/dto/order/OrderDto.java
package webapplication.dto.order;

import webapplication.dto.order.orderitem.OrderItemDto;
import webapplication.model.Order;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.Set;
import lombok.Data;

@Data
public class OrderDto {
    private Long id;
    private Long userId;
    private Set<OrderItemDto> orderItems;
    private LocalDate orderDate;
    private BigDecimal total;
    private Order.Status status;
}
